### PR TITLE
Pr/surface navigation

### PR DESCRIPTION
**Pull request by c4rlosviteri**
# Surface Navigation
## Overview
This PR introduces the Surface Navigation component, a specialized variant of Cedar's Surface component designed specifically for navigational elements. It provides comprehensive state management and styling capabilities for interactive surfaces.

## Key Features
- State-based styling for rest, hover, active, and visited states
- Configurable background colors, borders, shadows, and radii
- Accessibility-focused implementation with proper focus states
- TypeScript support with strict typing
- Modular SCSS architecture with mixins and variables

## Testing
- Component includes examples demonstrating various configurations
- Tested across all supported browsers
- Verified keyboard navigation and focus states
- Confirmed WCAG compliance for all states

## Documentation
- Added comprehensive documentation with usage examples
- Included TypeScript interfaces and prop definitions
- Documented accessibility considerations and best practices

## Usage Example
```vue
<CdrSurfaceNavigation 
  :background="{
    rest: 'primary',
    hover: 'secondary'
  }"
  :withBorder="{
    rest: false,
    hover: true
  }"
>
  <CdrSurfaceNavigationLink>
    Navigation Content
  </CdrSurfaceNavigationLink>
</CdrSurfaceNavigation>

### CODE REVIEW

File: src/components/surfaceNavigation/examples/SurfaceNavigation.vue [hunk 1]
```diff
@@ -0,0 +1,162 @@
...
+const handleFocus = (e: FocusEvent) => {
+  console.log('focused', e);
+}
+</script>
+
+<template>
+  <div class="example">
+    <h2>Surface Navigation</h2>
+    <template
+      v-for="{ exampleName, label, title, image, props, ratingProps, price, mediaObjectProps } in boxes"
```
Comment (mhewson) [inline src/components/surfaceNavigation/examples/SurfaceNavigation.vue:85]: I would be great to see a few more examples of this with random properties set, use this as our edge case test -- 

I would expect that for instance surface navigation without a media object - similar to fulfillment tile
nested surface navigations for some horrible situation gone wrong and so on

Comment (mhewson) [reply src/components/surfaceNavigation/examples/SurfaceNavigation.vue:85]: the updates are looking great, however I still would like to see enough examples to get a good sense of our edge cases

File: src/components/fulfillmentTile/styles/CdrFulfillmentTileIcon.module.scss [hunk 2]
```diff
@@ -1,26 +1,68 @@
-@import '../../../styles/settings/index';
-@import './vars/CdrFulfillmentTileIcon.vars';
-
 .cdr-fulfillment-tile-icon {
```
Comment (mhewson) [inline src/components/fulfillmentTile/styles/CdrFulfillmentTileIcon.module.scss:1]: this looks great, as a nit pick I would like to see these property value pairs be alphabetized

File: src/components/surface/CdrSurface.vue [hunk 3]
```diff
@@ -1,47 +1,30 @@
...
 
 const props = withDefaults(defineProps<surface>(), {
-  background: 'primary',
-  borderColor: 'primary',
-  borderStyle: 'solid',
-  borderWidth: 'sixteenth-x',
-  modifier: 'default',
-  radius: 'sharp',
-  shadow: 'flat',
+  variant: 'base',
```
Comment (mhewson) [inline src/components/surface/CdrSurface.vue]: is 'base' more appropriate here than 'default'?

File: src/components/surface/examples/Surface.vue [hunk 4]
```diff
@@ -26,7 +26,7 @@ const boxes: Example[] = [
     props: {
       radius: 'soft',
       background: 'secondary',
-      shadow: 'floating',
+      boxShadow: 'floating',
```
Comment (mhewson) [inline src/components/surface/examples/Surface.vue:29]: I think these should be named after the design foundations they implement rather than the css prop we use to expose them  - I think this one is "prominence"

File: src/components/surface/styles/CdrSurface.module.scss [hunk 5]
```diff
@@ -1,10 +1,198 @@
-@import '../../../styles/settings/index';
-@import './vars/CdrSurface.vars';
+@use "sass:map";
+
+// Property maps
+$properties: (
+  background: (
+    primary: var(--cdr-color-background-primary),
+    secondary: var(--cdr-color-background-secondary),
+    brand-spruce: var(--cdr-color-background-brand-spruce),
```
Comment (mhewson) [inline src/components/surface/styles/CdrSurface.module.scss]: lets rename this to just brand - brand-spruce seems really limiting from the context of what would happen via a palette being used

Comment (mhewson) [reply src/components/surface/styles/CdrSurface.module.scss]: which may change the way I go about setting the palettes up actually - rather than the primary color being mapped to a new value I may consider just updating the brand color for things like the membership palette

File: src/components/surface/styles/CdrSurface.module.scss [hunk 6]
```diff
@@ -1,10 +1,198 @@
...
+    sale: var(--cdr-color-background-sale)
+  ),
+  border-color: (
+    primary: var(--cdr-color-border-primary),
```
Comment (mhewson) [inline src/components/surface/styles/CdrSurface.module.scss:13]: anyway to do something like
allow the user to set a context, like primary, secondary, error - and then not have to define each prop unless its different than the generic context? 

so something like 

surface-secondary (this sets normal secondary values for background, border color, ect) and then
user also defines 
radius: round  or some such thing

File: src/components/surface/styles/CdrSurface.module.scss [hunk 7]
```diff
@@ -1,10 +1,198 @@
...
+    raised: var(--cdr-prominence-raised),
+    elevated: var(--cdr-prominence-elevated),
+    floating: var(--cdr-prominence-floating),
+    lifted: var(--cdr-prominence-lifted)
+  )
+);
+
+$states: (rest, hover, active, checked, loading, disabled, visited);
+$css-vars: (
+  background: --background,
```
Comment (mhewson) [inline src/components/surface/styles/CdrSurface.module.scss]: I wonder if we want the local variables to be something like --cdr-surface-background -- this way if someone ends up defining something like --background at their root scope we would not have variable naming collisions - additionally as these variables are still available to view in rendered code we have an easy signature "--cdr" to search for to see things like our overall usage

File: src/components/surface/styles/CdrSurface.module.scss [hunk 8]
```diff
@@ -1,10 +1,198 @@
...
-    $surface-border-colors,
-    $surface-component
-  );
+  appearance: none;
+  display: block;
+  position: relative;
+  text-align: left;
+
+  // Base CSS custom properties
+  --background: var(--cdr-surface-background, var(--cdr-surface-background-rest));
```
Comment (mhewson) [inline src/components/surface/styles/CdrSurface.module.scss]: we will want to add the sass vars in as a fallback for our sass only token importers -- or come up with a way to ensure they have that asset available

File: src/components/surface/styles/CdrSurface.module.scss [hunk 9]
```diff
@@ -1,10 +1,198 @@
...
+  // Generate base styles
+  @each $prop, $values in $properties {
+    @each $value-name, $value in $values {
+      &[data-#{$prop}='#{$value-name}'] {
+        #{map.get($css-vars, $prop)}: #{$value};
+      }
+    }
+  }
+
+  // Generate state-based styles
```
Comment (mhewson) [inline src/components/surface/styles/CdrSurface.module.scss:93]: this seems a bit fiddly 

how about something like 

@each $state in $states {
  @if $state != 'rest' {
    $selector: if($state == 'hover', 
                  '&:hover:not([aria-pressed="true"], [aria-checked="true"], :disabled)',
                  if($state == 'active', 
                     '&:active, &:focus',
                     '&[data-#{$state}="true"]'));
    
    @each $prop, $values in $properties {
      @each $value-name, $value in $values {
        &[data-#{$prop}-#{$state}='#{$value-name}'] {
          #{$selector} {
            #{map.get($css-vars, $prop)}: #{$value};
          }
        }
      }
    }
  }
}

File: src/components/surface/styles/CdrSurface.module.scss [hunk 10]
```diff
@@ -1,10 +1,198 @@
-@import '../../../styles/settings/index';
-@import './vars/CdrSurface.vars';
+@use "sass:map";
+
+// Property maps
```
Comment (mhewson) [inline src/components/surface/styles/CdrSurface.module.scss:3]: lets add a TODO comment to get this from cdr-tokens

File: src/components/surface/styles/CdrSurface.module.scss [hunk 11]
```diff
@@ -1,10 +1,198 @@
...
+  border-radius: var(--border-radius);
+  box-shadow: var(--box-shadow);
+  
+  &::before {
+    border: var(--border-width) var(--border-style) var(--border-color);
+    border-radius: inherit;
+    content: '';
+    inset: 0;
+    position: absolute;
+    transition: border-color var(--cdr-duration-3-x) var(--cdr-timing-function-ease);
```
Comment (mhewson) [inline src/components/surface/styles/CdrSurface.module.scss:124]: we should also use the will-change prop for selectors that will animate

File: src/components/surfaceNavigation/CdrSurfaceNavigationLink.vue [hunk 12]
```diff
@@ -0,0 +1,48 @@
+<script setup lang="ts">
```
Comment (mhewson) [inline src/components/surfaceNavigation/CdrSurfaceNavigationLink.vue]: should this use cdr-link as the base? if not it probably also needs to support 
:target="target"
:rel="computedRel"

File: src/utils/mapClasses.ts [hunk 13]
```diff
@@ -1,8 +1,18 @@
+/**
```
Comment (mhewson) [inline src/utils/mapClasses.ts:1]: nice! I appreciate the added docs

File: src/types/interfaces.ts [hunk 14]
```diff
@@ -2,12 +2,12 @@ import type { Component } from 'vue';
 import type {
   Tag,
   Space,
+  SpaceFixed,
```
Comment (mhewson) [inline src/types/interfaces.ts:5]: this is static rather than fluid space?

File: src/types/interfaces.ts [hunk 15]
```diff
@@ -2,12 +2,12 @@ import type { Component } from 'vue';
...
   SpaceOption,
   Shadow,
   Radius,
   BorderColor,
   BorderStyle,
-  Modifier,
```
Comment (mhewson) [inline src/types/interfaces.ts:10]: yes that was an oddball

File: src/types/interfaces.ts [hunk 16]
```diff
@@ -230,101 +230,85 @@ export interface Layout extends NameValuePair {
   containerName?: string;
 }
 
+interface StateConfig<T> {
```
Comment (mhewson) [inline src/types/interfaces.ts]: Where do you think we can capture out TS patterns and best practices like noting that we use T as a placeholder for another type that will be specified when StateConfig is used.  or are some things just a generic convention?

**Review by mhewson — APPROVED** [review]

File: src/components/fulfillmentTile/styles/CdrFulfillmentTileIcon.module.scss [hunk 17]
```diff
@@ -1,14 +1,14 @@
...
-  width: 2.2rem;
-  height: 2.2rem;
-  background: white;
-  display: flex;
   align-items: center;
-  justify-content: center;
+  background: white;
```
Comment (mhewson) [inline src/components/fulfillmentTile/styles/CdrFulfillmentTileIcon.module.scss]: should be --cdr-color-background-primary?

File: src/components/fulfillmentTile/styles/CdrFulfillmentTileIcon.module.scss [hunk 18]
```diff
@@ -1,14 +1,14 @@
...
   border-radius: var(--cdr-radius-round);
-  position: relative;
+  display: flex;
   fill: var(--cdr-fulfillment-tile-icon-fill);
   flex-shrink: 0;
+  height: 2.2rem;
```
Comment (mhewson) [inline src/components/fulfillmentTile/styles/CdrFulfillmentTileIcon.module.scss:8]: Note for future -- We have Icon spacing implemented in several components - I want to centralize this logic so the icon sizes stay consistent -- look at banner and icon button as examples

File: src/components/fulfillmentTile/styles/CdrFulfillmentTileIcon.module.scss [hunk 19]
```diff
@@ -1,14 +1,14 @@
...
+  justify-content: center;
+  position: relative;
+  width: 2.2rem;
```
Comment (mhewson) [inline src/components/fulfillmentTile/styles/CdrFulfillmentTileIcon.module.scss:11]: also token

File: src/components/fulfillmentTile/styles/CdrFulfillmentTileIcon.module.scss (hunk 17 above)
Comment (mhewson) [reply src/components/fulfillmentTile/styles/CdrFulfillmentTileIcon.module.scss]: or data-background="primary"
